### DISCUSSION TITLE
Copy the output

### DISCUSSION THREAD

**Discussion by isaacwein** [x1unix/go-playground#340] [Q&A]
is there a way to copy the output?
And why is there an option to save it as an image?
And what the image that got saved?
![image](https://github.com/x1unix/go-playground/assets/34663972/8444ed71-2008-4b1d-bb4e-e313504c922c)

saved image 👇
![download](https://github.com/x1unix/go-playground/assets/34663972/36f7bf65-cd6f-4e24-80ca-db62ec67735a)

**Comment by x1unix** [comment]
@isaacwein 

> is there a way to copy the output?

* on computer - select text with mouse and use Ctrl+C (or cmd+C)
* on tablet/phone - there is a copy button that appear when you click on output:

<img width="398" alt="image" src="https://github.com/x1unix/go-playground/assets/9203548/33e0465d-3172-4eb4-995d-eba082243532">

> And what the image that got saved?

Because terminal is based on xterm.js which is hardware-accelerated and rendered on Canvas.
Hardware acceleration is implemented for shell escape symbols as control commands, colors, etc.

You can switch back to DOM implementation in the Settings window.

<img width="523" alt="image" src="https://github.com/x1unix/go-playground/assets/9203548/b0745d31-d15b-473e-8f99-a52c53f7282d">

**Comment by isaacwein** [comment]
i witched to DOM but there is no option to copy

![image](https://github.com/x1unix/go-playground/assets/34663972/103d48d9-ed8d-4a31-9d78-ad7a9f794a99)


also ctrl+C is giving me the image not the output text

**Reply by x1unix** [reply]
Can't reproduce
regarding right click context menu - already answered above.

https://github.com/x1unix/go-playground/assets/9203548/cf7af659-84bc-487a-85c9-25b5f093c728
